iceoryx_binding_c/test/moduletests/test_chunk.cpp

```diff
@@ -159,4 +159,60 @@ TEST_F(Chunk_test, GettingUserHeaderFromConstChunkHeaderWorks)
+    constexpr uint32_t USER_PAYLOAD_SIZE(42U);
+    constexpr uint32_t USER_PAYLOAD_ALIGNMENT(64U);
+    constexpr uint32_t USER_HEADER_SIZE = 16U;
+    constexpr uint32_t USER_HEADER_ALIGNMENT = 8U;
+    void* userPayload{nullptr};
+    ASSERT_EQ(iox_pub_loan_aligned_chunk_with_user_header(publisher,
+                                                          &userPayload,
+                                                          USER_PAYLOAD_SIZE,
+                                                          USER_PAYLOAD_ALIGNMENT,
+                                                          USER_HEADER_SIZE,
+                                                          USER_HEADER_ALIGNMENT),
+              AllocationResult_SUCCESS);
+
+    const iox_chunk_header_t* chunkHeader = iox_chunk_header_from_user_payload(userPayload);
+
+    ASSERT_EQ(iox_chunk_header_user_chunk_size(chunkHeader), USER_HEADER_SIZE + USER_PAYLOAD_SIZE);
+    ASSERT_EQ(iox_chunk_header_user_header_size(chunkHeader), USER_HEADER_SIZE);
+    ASSERT_EQ(iox_chunk_header_user_payload_size(chunkHeader), USER_PAYLOAD_SIZE);
+    ASSERT_EQ(iox_chunk_header_user_payload_alignment(chunkHeader), USER_PAYLOAD_ALIGNMENT);
+}
+
+TEST_F(Chunk_test, GettingSequenceNumberFromConstChunkHeaderWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "fd564011-77a7-494a-93ae-f5b3f895cf95");
+    constexpr uint32_t USER_PAYLOAD_SIZE(42U);
+    constexpr uint32_t USER_PAYLOAD_ALIGNMENT(64U);
+    constexpr uint32_t USER_HEADER_SIZE = 16U;
+    constexpr uint32_t USER_HEADER_ALIGNMENT = 8U;
+
+    void* userPayload1{nullptr};
+    ASSERT_EQ(iox_pub_loan_aligned_chunk_with_user_header(publisher,
+                                                          &userPayload1,
+                                                          USER_PAYLOAD_SIZE,
+                                                          USER_PAYLOAD_ALIGNMENT,
+                                                          USER_HEADER_SIZE,
+                                                          USER_HEADER_ALIGNMENT),
+              AllocationResult_SUCCESS);
+    const iox_chunk_header_t* chunkHeader1 = iox_chunk_header_from_user_payload(userPayload1);
+
+    void* userPayload2{nullptr};
+    ASSERT_EQ(iox_pub_loan_aligned_chunk_with_user_header(publisher,
+                                                          &userPayload2,
+                                                          USER_PAYLOAD_SIZE,
+                                                          USER_PAYLOAD_ALIGNMENT,
+                                                          USER_HEADER_SIZE,
+                                                          USER_HEADER_ALIGNMENT),
+              AllocationResult_SUCCESS);
+
+    const iox_chunk_header_t* chunkHeader2 = iox_chunk_header_from_user_payload(userPayload2);
+
+    ASSERT_GT(iox_chunk_header_sequence_number(chunkHeader2), iox_chunk_header_sequence_number(chunkHeader1));
```

Comment:
Similar here, although it would be a whitebox test, it might be better to just cast the `chunkHeader` to a `ChunkHeader` and call the `sequenceNumber` method.

@elfenpiff what do you think?

Reply:
I agree! Yes it would be somehow a whitebox test but in the end the C API should forward every call to the c++ API and should never contain any kind of logic. The C++ API is well tested therefore we can assume it works as expected so why not use it here.